File: factorial.php
<?php
?>
<!DOCTYPE html>
<html>
<head>
    <title>Calcular Factorial</title>
</head>
<body>
    <h1>Calculadora de Factorial</h1>
    
    <form method="POST">
        <label>Número:</label>
        <input type="number" name="numero" min="0" required>
        <br><br>
        
        <button type="submit">Calcular</button>
    </form>
    
    <?php
    if ($_SERVER['REQUEST_METHOD'] == 'POST') {
        $n = intval($_POST['numero']);
        
        if ($n < 0) {
            echo "<h2>Error: El número no puede ser negativo</h2>";
        } else {
            $resultado = 1;
            for ($i = 2; $i <= $n; $i++) {
                $resultado *= $i;
            }
            echo "<h2>Resultado: $n! = $resultado</h2>";
        }
    }
    ?>
    
    <br>
    <a href="index.php">Volver al inicio</a>
</body>
</html>

File: factorial.py
"""
Módulo para calcular el factorial de un número
"""

def calcular_factorial(n):
    """
    Calcula el factorial de un número
    
    Args:
        n: Número entero no negativo
        
    Returns:
        int: El factorial de n
        
    Raises:
        ValueError: Si n es negativo
    """
    if n < 0:
        raise ValueError("El número no puede ser negativo")
    
    if n == 0 or n == 1:
        return 1
    
    resultado = 1
    for i in range(2, n + 1):
        resultado *= i
    
    return resultado


def main():
    """Función principal para calcular factorial"""
    print("\n" + "=" * 40)
    print("CALCULADORA DE FACTORIAL")
    print("=" * 40)
    
    try:
        num = int(input("Ingrese un número entero positivo: "))
        
        if num < 0:
            print("Error: Ingrese un número no negativo")
        else:
            resultado = calcular_factorial(num)
            print(f"\n{num}! = {resultado}")
            print("=" * 40 + "\n")
            
    except ValueError:
        print("Error: Ingrese un número entero válido")


if __name__ == "__main__":
    main()
